### PR TITLE
[ML-28918] Split terraform-cd into two actions: one for default branch, one for release branch

### PR DESCRIPTION
**Pull request by zhe-db**
Split terraform-cd into two actions: one for default branch terraform deployment and one for release branch terraform deployment.

Manual test with CUJ:
https://github.com/databricks/mlops-azure-cuj/pull/75
- Changed `run_mode`in `staging/training-job,tf` and `prod/training-job.tf` in main branch, which only triggers `staging` job that only updates the `run_mode` in staging workspace, the prod workspace one did not get changed
- Filed a PR to merge main into release, which triggers `prod` job, `run_mode` in prod workspace is updated.

### CODE REVIEW

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/.github/workflows/terraform-cd-prod.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# This GitHub workflow deploys Terraform resources (ML resource config and more)
+# defined under {{cookiecutter.project_name}}/terraform, when PRs are merged into the default branch
```
Comment (mingyu89) [inline {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/.github/workflows/terraform-cd-prod.yml:2]: nit `when PRs are merged into the default branch` => `when PRs are merged into the release branch`

**Review by mingyu89 — APPROVED** [review]

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/.github/workflows/terraform-cd-prod.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
+# This GitHub workflow deploys Terraform resources (ML resource config and more)
+# defined under {{cookiecutter.project_name}}/terraform, when PRs are merged into the release branch
```
Comment (arpitjasa-db) [inline {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/.github/workflows/terraform-cd-prod.yml:2]:
```suggestion
# defined under {{cookiecutter.project_name}}/terraform/prod, when PRs are merged into the release branch
```

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/.github/workflows/terraform-cd-staging.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # This GitHub workflow deploys Terraform resources (ML resource config and more)
 # defined under {{cookiecutter.project_name}}/terraform, when PRs are merged into the default branch
```
Comment (arpitjasa-db) [inline {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/.github/workflows/terraform-cd-staging.yml:2]:
```suggestion
# defined under {{cookiecutter.project_name}}/terraform/staging, when PRs are merged into the default branch
```

File: {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name}}/terraform/README.md [hunk 4]
```diff
@@ -51,7 +51,7 @@ expected remote state backend and store credentials in CI/CD.
 Alternatively, you can follow
 [Terraform docs](https://developer.hashicorp.com/terraform/language/settings/backends/configuration) to specify a custom remote state
 backend in `staging/provider.tf` and `prod/provider.tf`, store credentials for accessing the backend in CI/CD,
-and configure the `.github/workflows/terraform-ci.yml` and `.github/workflows/terraform-cd.yml` workflows
+and configure the `.github/workflows/terraform-ci.yml` and `.github/workflows/terraform-cd-prod.yml` and `.github/workflows/terraform-cd-staging.yml` workflows
```
Comment (arpitjasa-db) [inline {{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name}}/terraform/README.md:54]:
```suggestion
and configure the `.github/workflows/terraform-ci.yml`, `.github/workflows/terraform-cd-prod.yml`, and `.github/workflows/terraform-cd-staging.yml` workflows
```

**Review by arpitjasa-db — APPROVED** [review]

**Pull request merged**